samples/net/ipv4_autoconf/README.rst

```diff
@@ -0,0 +1,69 @@
+.. _ipv4-autoconf-sample:
+
+Sample IPv4 autoconf client application
+#######################################
+
+Overview
+********
+
+This application starts a IPv4 autoconf and self-assigns
+a random IPv4 address in the 169.254.0.0/16 range, it defends
```

Comment:
@jukkar, @dbkinder : FYI, I think the documentation for this sample needs cleanup. E.g. I doubt "defends" is the right word here.

Reply:
@pfalcon, wording matches with RFC descriptions, should be ok as is 
see https://tools.ietf.org/html/rfc3927#section-2

Reply:
@locomuco : Thanks for response, and indeed, interesting wording is in the RFC ;-). I'd still think that outside that RFC, it sounds a bit strange. There're few other minor issues with the docs, which I didn't mark explicitly but meant when talked about need for cleanup.